subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenScopesTestIntegTest.groovy

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.maven.MavenFileModule
+import org.gradle.test.fixtures.maven.MavenJavaModule
+import org.gradle.test.fixtures.maven.MavenModule
+
+class MavenScopesTestIntegTest extends AbstractIntegrationSpec {
+    def "test adding custom variant with dependency mapped to Maven runtime scope"() {
+        given:
+        settingsFile << "rootProject.name = 'publishTest'"
+        buildFile << """
+            apply plugin: 'maven-publish'
```

Comment:
Nitpick, but shouldn't new tests use the `plugins` block?

Reply:
Yep, and the duplicated test setup can be consolidated.